sdks/go/pkg/beam/core/sdf/sdf.go

```diff
@@ -96,3 +98,12 @@ type WatermarkEstimator interface {
 	// splits or checkpoints to advance the output watermark of the restriction's stage.
 	CurrentWatermark() time.Time
 }
+
+// ToWatermarkEstimator is an interface used to represent a user defined watermark estimator that
+// has the ability to observe timestamps of elements outputted from a ParDo's emit function.
+type ToWatermarkEstimator interface {
```

Comment:
If TO is supposed to be an acronym for TimestampObserving, please capitalize it accordingly.  
I'd also recommend simplifying it to `TimestampObserver`. Yes it only applies to WatermarkEstimators with the embed, but that can be clarified in the type documentation, rather than the whole name.